onnxscript/rewriter/onnx_fusions/_layer_norm.py

```diff
@@ -0,0 +1,128 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+from __future__ import annotations
+
+import onnx_ir as ir
+
+from onnxscript.rewriter import _fusion_utils, _ir_utils, pattern
+
+"""
+Layer Normalization fusion optimization.
+
+This module contains rewrite rules for fusing Layer Normalization patterns into the
+ONNX LayerNormalization operator.
+
+Layer Normalization performs normalization over the last D dimensions as specified by the axis.
+The computation follows: Y = scale * (X - mean) / sqrt(variance + epsilon) + bias
+
+Key points for the fusion optimization:
+* Following restrictions from opset 17 LayerNormalization:
+* Input, scale, and bias must be of same type T in {float16, bfloat16, float, double}
+* The normalization can be done in a different precision than the input type (bfloat16 or float),
+which is also the precision of the output mean/invstddev
+"""
+
+float_types = frozenset(
+    [
+        ir.DataType.FLOAT,
+        ir.DataType.FLOAT16,
+        ir.DataType.BFLOAT16,
+        ir.DataType.DOUBLE,
+    ]
+)
+fp_float_types = frozenset([ir.DataType.FLOAT, ir.DataType.DOUBLE])
```

Comment:
```suggestion
FLOAT_TYPES = frozenset(
    [
        ir.DataType.FLOAT,
        ir.DataType.FLOAT16,
        ir.DataType.BFLOAT16,
        ir.DataType.DOUBLE,
    ]
)
HIGH_PRECISION_FLOAT_TYPES = frozenset([ir.DataType.FLOAT, ir.DataType.DOUBLE])
```

nit: upper-case for constants

Reply:
Done. I also gave them more meaningful names.